File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
       int n = nums.length;
       int maxele = nums[0];
       int currentSum = nums[0];
       for(int i = 1 ; i<n; i++){
            currentSum = Math.max(nums[i],nums[i]+currentSum);
            maxele  = Math.max(currentSum,maxele); 
       } 
       return (int) maxele;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int maxx = INT_MIN;
        int sum = 0;
        for(int i = 0; i<n;i++){
            sum += nums[i];
            if(sum>maxx){
                maxx = sum;
            }
            if(sum<0){
                sum = 0;
            }
        }
        return maxx;
    }
};